types/gulp-modernizr/index.d.ts

```diff
@@ -0,0 +1,70 @@
+// Type definitions for gulp-modernizr 1.0
+// Project: https://github.com/doctyper/gulp-modernizr
+// Definitions by: remisery <https://github.com/remisery>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare function GulpModernizr(fileOrParams?: string | GulpModernizr.Params, parames?: GulpModernizr.Params): any;
+
```

Comment:
Apart that after looking at your code, It seems you want to do
```TypeScript
import stream = require('stream');
``` 
and then use ```stream.Transform``` as your return type for this function.